README.md

```diff
@@ -82,6 +82,20 @@ See the [Reporter Interface section of the API documentation](https://github.com
 
 Looking for more examples? Check out the [examples folder](https://github.com/hapijs/good/tree/master/examples).
 
+## Reporter Type
```

Comment:
The thing is that as it is now. the user also needs to know that this might cause an issue since he is the one defining the pipeline.
I think the best solution would be if good will check the interface of each stream and see if the pipeline was defined correctly. 
To prevent breaking changes  good can show a warning that the pipeline doesn't make sense but throwing an error is probably preferred.
This also solves the issue mentioned in the previous comment since if good is going to warn when weird pipelines are defined it is not as necessary to give streams a clear name indicating their role in the pipeline